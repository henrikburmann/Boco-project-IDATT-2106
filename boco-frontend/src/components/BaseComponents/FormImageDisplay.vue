<template>
  <!-- Image -->
  <img
    :src="image"
    class="w-2/5 inline"
    alt="Bilde av gjenstanden"
    @click="toggleModal"
  />
  <!-- Modal -->
  <CustomFooterModal
    @close="toggleModal"
    :message="'Bilder som fjernes kan ikke lastes opp på nytt uten å laste opp et annet bilde først.'"
    :title="'Er du sikker på at du vil fjene bilde?'"
    :visible="show"
  >
    <!-- Buttons for modal footer -->
    <div class="flex justify-center p-2">
      <ColoredButton :text="'Avbryt'" @click="toggleModal" class="m-2" />
      <ColoredButton
        :text="'Slett'"
        :color="'red'"
        @click="removeImage"
        class="m-2"
      />
    </div>
  </CustomFooterModal>
</template>

<script>
import ColoredButton from "@/components/BaseComponents/ColoredButton";
import CustomFooterModal from "@/components/BaseComponents/CustomFooterModal.vue";

/**
 * Displays uploaded image for forms.
 */
export default {
  components: {
    ColoredButton,
    CustomFooterModal,
  },
  data() {
    return {
      show: false,
    };
  },
  props: {
    image: String,
  },
  methods: {
    /**
     * Emits remove image to parent
     */
    removeImage() {
      this.$emit("remove");
    },
    /**
     * Toggles modal
     */
    toggleModal() {
      this.show = !this.show;
    },
  },
};
</script>
