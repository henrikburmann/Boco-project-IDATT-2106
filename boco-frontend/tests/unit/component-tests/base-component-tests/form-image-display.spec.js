import { mount } from "@vue/test-utils";
import FormImageDisplay from "@/components/BaseComponents/FormImageDisplay.vue";

describe("FormImageDisplay component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(FormImageDisplay, {
      //passing prop to component
      props: {
        image: "http://localhost:3000/api/images/5",
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
