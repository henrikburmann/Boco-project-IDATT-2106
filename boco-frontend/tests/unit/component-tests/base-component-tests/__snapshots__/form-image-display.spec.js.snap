// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormImageDisplay component renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- Image -->
  <img
    alt="Bilde av gjenstanden"
    class="w-2/5 inline"
    src="http://localhost:3000/api/images/5"
  />
  <!-- Modal -->
  
  <!-- Main modal -->
  <!--v-if-->
  
  
</div>
`;
